packages/router/src/router.tsx

```diff
@@ -185,54 +173,95 @@ const Router: React.FC<RouterProps> = ({
   )
 }
 
+/**
+ * Find the active (i.e. first matching) route and discard any other routes.
+ * Also, keep any <Set>s wrapping the active route.
+ */
+function activeRouteTree(
```

Comment:
Nice, I was very very scared of reviewing this, but your comments and the code organization made this very easy, thanks Tobbe!